lib/reek/code_comment.rb

```diff
@@ -64,19 +63,77 @@ def sanitized_comment
         gsub(SANITIZE_REGEX, ' ').
         strip
     end
+  end
+
+  #
+  # A typical configuration via code comment looks like this:
+  #
+  #   :reek:DuplicateMethodCall { enabled: false }
+  #
+  # There are a lot of ways a user can introduce some errors here:
+  #
+  # 1.) Unknown smell detector
+  # 2.) Garbage in the detector configuration like { thats: a: bad: config }
+  # 3.) Unknown configuration keys (e.g. by doing a simple typo: "exclude" vs. "exlude" )
+  # 4.) Bad data types given as values for those keys
+  # This class validates [1] and [2] at the moment but will also validate [3]
+  # and [4] in the future.
+  #
+  # :reek:TooManyInstanceVariables: { max_instance_variables: 7 }
+  class CodeCommentValidator
```

Comment:
I wanted to stress that CodeCommentValidator is **only** used by CodeComment.
What do you think about modelling as pseudo-inner class a la:

```Ruby
class CodeComment
  # ...
  class CodeCommentValidator
  end
end
```

This wont prevent anybody from instantiating it but at least it communicates our intent here in a pretty clear way.